x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/InferenceResultFieldMapper.java

```diff
@@ -148,10 +148,15 @@ public InferenceResultFieldMapper() {
 
     @Override
     protected void parseCreateField(DocumentParserContext context) throws IOException {
-        XContentParser parser = context.parser();
-        failIfTokenIsNot(parser, XContentParser.Token.START_OBJECT);
-
-        parseAllFields(context);
+        boolean withinLeafObject = context.path().isWithinLeafObject();
+        try {
+            context.path().setWithinLeafObject(true);
```

Comment:
This is necessary to avoid dot expansion in parsers, and thus avoid traversing structures like:

```json
"base_field": {
    "multi_field": {
    }
}
```

Instead, we get a dotted representation of the field:

```json
"base_field.multi_field": {
}
```

Reply:
I'll do that 👍